Exercice_2_-_Le_chiffre_de_Cesar/caesar.py

```diff
@@ -80,83 +80,78 @@
 >>> chr(90+3)
 ']'
 """
-
+import string
 
 def do_cipher(plain_text: str, key: int) -> str:
-    # Met ton code ici
-    return "Il faudrait retourner un vrai truc ;)"
+    LIM_MIN_DOWN = ord('A')
+    LIM_MIN_UP = ord('Z')
+    ALPHABET_LOOP = len(string.ascii_uppercase)
+    
+    cipher_text = ""
 
+    plain_text = plain_text.upper()
+    
+    for i in plain_text:
+        if ord(i) in range(ord('A'),ord('Z')+1):
```

Comment:
En fait tu te complique un peu la vie ici... tu veut savoir si la lettre est une lettre majuscule. Or tu as justement une chaîne avec toute les majuscules. Autant l'utiliser : `if i in string.ascii_uppercase:` c'est quand même vachement plus lisible, non ?